File: mathenger-api/src/main/java/com/example/mathengerapi/models/GroupChat.java
package com.example.mathengerapi.models;

import com.example.mathengerapi.models.enums.ChatType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Entity
@Data
@DiscriminatorValue(value = "GROUP_CHAT")
public class GroupChat extends Chat {
    @Column(length = 40)
    private String name;
    @Column(nullable = false, length = 15)
    private String color;
    @Transient
    private final ChatType chatType = ChatType.GROUP_CHAT;
    @ManyToOne
    @JoinColumn(name = "creator_id")
    private Account creator;
    @ManyToMany
    @JoinTable(name = "chat_admin",
            joinColumns = {@JoinColumn(name = "chat_id", referencedColumnName = "id")},
            inverseJoinColumns = {@JoinColumn(name = "admin_id", referencedColumnName = "id")})
    private List<Account> admins;
}

File: mathenger/Mathenger/Mathenger/models/GroupChat.cs
using System.Collections.ObjectModel;
using Mathenger.Models.Enums;
using Newtonsoft.Json;

namespace Mathenger.models
{
    public class GroupChat : Chat
    {
        [JsonProperty("name")] public string Name { get; set; }
        [JsonProperty("color")] public string Color { get; set; }
        [JsonProperty("creator")] public Account Creator { get; set; }
        [JsonProperty("admins")] public ObservableCollection<Account> Admins { get; set; }
        public override void Update(Chat chat)
        {
            base.Update(chat);
            if (chat.ChatType == ChatType.GROUP_CHAT)
            {
                var newChat = chat as GroupChat;
                Admins = newChat.Admins;
                Name = newChat.Name;
                Color = newChat.Color;
                Creator = newChat.Creator;
            }
        }
    }
}